ic_routing_board_generation/board_generator/board_generator_v3_1_0_rb.py

```diff
@@ -0,0 +1,476 @@
+# constants has been moved to jumanji/environments/combinatorial/routing/constants.py
+# import jumanji.environments.combinatorial.routing.constants
+#from jumanji.environments.combinatorial.routing.constants import TARGET, HEAD, EMPTY
+#from jumanji.environments.combinatorial.routing.constants import SOURCE as WIRE
+HEAD, TARGET, WIRE, EMPTY = 4,3,2,0
+# Also available to import from constants OBSTACLE, NOOP, LEFT, LEFT, UP, RIGHT, DOWN
+from dataclasses import dataclass
+import numpy as np
+from copy import deepcopy
+import random
+from typing import List, Tuple
+#from jax.numpy import asarray # Currently jaxlib is not supported on windows.  This will have to be sorted.
+#from env_viewer import RoutingViewer # Currently jaxlib is not supported on windows.  This will have to be sorted.
+
+@dataclass#
+class Position:
+    # Class of 2D tuple of ints, indicating the size of an array or a 2D position or vector.
+    x: int
+    y: int
+
+class Board:
+    """ The boards are 2D np.ndarrays of wiring routes on a printed circuit board.
+
+    The coding of the boards is as follows:
+    Empty cells are coded 0.
+    Obstacles are coded 1.
+    Heads are encoded starting from 4 in multiples of 3: 4, 7, 10, ...
+    Targets are encoded starting from 3 in multiples of 3: 3, 6, 9, ...
+    Wiring routes connecting the head/target pairs are encoded starting
+            at 2 in multiples of 3: 2, 5, 8, ...
+
+    Args:
+        rows, cols (int, int) : Dimensions of the board.
+
+    """
+    def __init__(self, rows: int, cols: int):
```

Comment:
should take num_agents in init

Reply:
Done.
Let's please revisit this with the group.